### ISSUE TITLE
Compatibility certification request for EE4J implementation of Jakarta RESTful Web Services

### ISSUE THREAD

**Issue by jansupol** [eclipse-ee4j/jaxrs-api#789]
- [x] Organization Name ("Organization") and, if applicable, URL
 * Eclipse Foundation
- [x] Product Name, Version and download URL (if applicable)
 * Jersey 2.28
   * [jersey-bean-validation](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/jersey-bean-validation/2.28/jersey-bean-validation-2.28.jar)
   * [jersey-cdi1x](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/cdi/jersey-cdi1x/2.28/jersey-cdi1x-2.28.jar)
   * [jersey-cdi1x-servlet](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/cdi/jersey-cdi1x-servlet/2.28/jersey-cdi1x-servlet-2.28.jar)
   * [jersey-cdi1x-transaction](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/cdi/jersey-cdi1x-transaction/2.28/jersey-cdi1x-transaction-2.28.jar)
   * [jersey-client](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/core/jersey-client/2.28/jersey-client-2.28.jar)
   * [jersey-common](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/core/jersey-common/2.28/jersey-common-2.28.jar)
   * [jersey-container-grizzly2-http](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/containers/jersey-container-grizzly2-http/2.28/jersey-container-grizzly2-http-2.28.jar)
   * [jersey-container-servlet](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/containers/jersey-container-servlet/2.28/jersey-container-servlet-2.28.jar)
   * [jersey-container-servlet-core](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/containers/jersey-container-servlet-core/2.28/jersey-container-servlet-core-2.28.jar)
   * [jersey-entity-filtering](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/jersey-entity-filtering/2.28/jersey-entity-filtering-2.28.jar)
   * [jersey-gf-ejb](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/containers/glassfish/jersey-gf-ejb/2.28/jersey-gf-ejb-2.28.jar)
   * [jersey-hk2](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/inject/jersey-hk2/2.28/jersey-hk2-2.28.jar)
   * [jersey-media-jaxb](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-jaxb/2.28/jersey-media-jaxb-2.28.jar)
   * [jersey-media-json-binding](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-json-binding/2.28/jersey-media-json-binding-2.28.jar)
   * [jersey-media-json-jackson](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-json-jackson/2.28/jersey-media-json-jackson-2.28.jar)
   * [jersey-media-json-jettison](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-json-jettison/2.28/jersey-media-json-jettison-2.28.jar)
   * [jersey-media-json-processing](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-json-processing/2.28/jersey-media-json-processing-2.28.jar)
   * [jersey-media-moxy](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-moxy/2.28/jersey-media-moxy-2.28.jar)
   * [jersey-media-multipart](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-multipart/2.28/jersey-media-multipart-2.28.jar)
   * [jersey-media-sse](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/media/jersey-media-sse/2.28/jersey-media-sse-2.28.jar)
   * [jersey-mvc](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/jersey-mvc/2.28/jersey-mvc-2.28.jar)
   * [jersey-mvc-jsp](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/ext/jersey-mvc-jsp/2.28/jersey-mvc-jsp-2.28.jar)
   * [jersey-server](https://oss.sonatype.org/content/groups/public/org/glassfish/jersey/core/jersey-server/2.28/jersey-server-2.28.jar)
- [x] Specification Name, Version and download URL
 * [Jakarta RESTful Web Services 2.1](https://oss.sonatype.org/content/groups/public/jakarta/ws/rs/jakarta.ws.rs-api/2.1.6/jakarta.ws.rs-api-2.1.6.jar)
- [x] TCK Version, digital SHA-256 fingerprint and download URL
 * [Jakarta RESTful Web Services TCK 2.0](http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-restful-ws-tck-2.1.0.zip) 
 * SHA-256: 64846e2297e5efad03e02c1ff9003b4e7705088699c513e425b7290c66a42630
- [x] Public URL of TCK Results Summary
 * [TCK results summary]( https://eclipse-ee4j.github.io/jersey/tck/jakarta-restful-web-services-2.1)
- [x] Any Additional Specification Certification Requirements
 * None
- [x] Java runtime used to run the implementation
 * Oracle JDK 1.8.0_191
- [x] Summary of the information for the certification environment, operating system, cloud, ...
 *  NAME="Alpine Linux" ID=alpine VERSION_ID=3.8.1 PRETTY_NAME="Alpine Linux v3.8"
- [x] By checking this box I acknowledge that the Organization I represent accepts the terms of the [EFTL](https://www.eclipse.org/legal/tck.php).
- [x] By checking this box I attest that all TCK requirements have been met, including any compatibility rules.

Tests results:

```
 [javatest.batch] Completed running 2663 tests.
 [javatest.batch] Number of Tests Passed      = 2663
 [javatest.batch] Number of Tests Failed      = 0
 [javatest.batch] Number of Tests with Errors = 0
```

**Comment by jansupol** [comment]
(Influenced by [JavaMail example](https://github.com/eclipse-ee4j/mail-spec/issues/1))

**Comment by mkarg** [comment]
What actually do you want the committers of JAX-RS to do with this issue?

**Comment by jansupol** [comment]
> What actually do you want the committers of JAX-RS to do with this issue?

This is a request for the implementation to run the TCK with the API. The reason why this is an issue in the JAX-RS and not at the implementation is not quite clear to me. Maybe it could have been grabbed by any implementation, or maybe it is for the JAX-RS project to have the certification process covered on its own issue tracker. I take it as a document that is filled for the release review.

So I need some committer of JAX-RS to run the TCK. I assume that someone would be me.

**Comment by mkarg** [comment]
@jansupol I share your point of view. ;-)

**Comment by starksm64** [comment]
That is outlined in the https://github.com/jakartaee/specification-committee/blob/master/process.adoc document. The specification project members are responsible for vetting the validity of the certification request. See the Certification Resolution section of that Jakarta EE TCK Process 1.0 document.

**Comment by spericas** [comment]
+1

**Comment by mkarg** [comment]
@jansupol @spericas I wonder if this request can be processed due to formal reasons: Neither JAX-RS 2.1.6 nor the TCK is officially released. So how can we certify compliance?

**Comment by bshannon** [comment]
This looks good, although the TCK results summary needs to duplicate all the information from this certification request.  Usually we've just been copying and pasting the content to the top of the TCK results page.  See for example the [Jakarta Mail TCK results page](https://eclipse-ee4j.github.io/mail/TCK-Results).

**Comment by arjantijms** [comment]
+1

**Comment by chkal** [comment]
+1

**Comment by asoldano** [comment]
+1

**Comment by m0mus** [comment]
+1

**Comment by Tomas-Kraus** [comment]
+1

**Comment by jansupol** [comment]
> I wonder if this request can be processed due to formal reasons: Neither JAX-RS 2.1.6 nor the TCK is officially released. So how can we certify compliance?

We have the API in staging, we have the TCK available in Eclipse download page. Once this is approved and the API is released then everybody will see the API and the TCK that is used to test the compliance. The process is just being created and is new to me as well, if you can see any flaws in the process there is the Specification committee and Steering committee to give the most relevant answers.

**Comment by kwsutter** [comment]
@jansupol Could you please update the TCK link on this CR and the TCK-Results page to point at the official location of the TCK?  https://download.eclipse.org/jakartaee/restful-ws/2.1/?d. Thanks!

**Comment by mkarg** [comment]
@kwsutter There is an *official* location for the Jakarta REST TCK? What is our technical way to push our binaries there? Maybe it's just me, but I cannot remember that the Jakarta REST committers had been officially informed about that *official* location. Can someone please point me to that information, seems I missed that.

**Comment by bshannon** [comment]
Yes, there's an official location where Eclipse will publish the approved and digitally signed
versions of the TCKs that can be used for making claims of compatibility.  I don't remember
all the places we've described this, but it's in the [spec review checklist](https://github.com/jakartaee/specification-committee/blob/master/spec_review_checklist.md) and mentioned with
less detail in the [TCK process document](https://github.com/jakartaee/specification-committee/blob/master/process.adoc).  Once the spec is approved, a member of the spec
committee will copy the TCK from the URL included in the spec Pull Request to the official
download location and sign it.  It's this official download location that will be included in the
web pages describing the approved spec.

**Comment by mkarg** [comment]
Uhm... what is a "*Spec* Pull Request"?

**Comment by spericas** [comment]
@mkarg https://github.com/jakartaee/specifications/pull/7

**Comment by jansupol** [comment]
Approved by [Lazy Consensus](https://github.com/jakartaee/specification-committee/blob/master/process.adoc#certification-resolution) and can be closed.

**Comment by edbratt** [comment]
Revised the d/l link to point at final TCK URL.
